packages/apollo-voyager-metrics/src/index.ts

```diff
@@ -33,64 +31,138 @@ const serverResponseMetric = new Prometheus.Histogram({
   labelNames: ['request_type', 'error']
 })
 
-export function getMetrics (req: IncomingMessage, res: Response) {
-  res.set('Content-Type', Prometheus.register.contentType)
-  res.end(Prometheus.register.metrics())
+interface ResolverObject {
+  [key: string]: IFieldResolver<any, any>
+}
 
-  resolverTimingMetric.reset()
-  resolverRequestsMetric.reset()
-  serverResponseMetric.reset()
+export function enableDefaultMetricsColleciton () {
+  if (!promMetricsEnabled) {
+    Prometheus.collectDefaultMetrics()
+    promMetricsEnabled = true
+  }
 }
 
-export function responseLoggingMetric (req: IncomingMessage, res: ResponseWithVoyagerMetrics, next: NextFunction) {
-  const requestMethod = req.method as string
+export function wrapResolversForMetrics (resolvers: {[key: string]: ResolverObject}): {[key: string]: ResolverObject} {
+  const output: {[key: string]: ResolverObject} = {}
+  for (const typeKey in resolvers) {
```

Comment:
I find this code slightly hard to understand. Between the typescript types and the `for ... in` and `continue` statements, I find this quite syntax heavy. 

I think maybe it might be worth renaming `resolvers` to `resolverMappings`

And then doing something like this.

```js
const typeKeys = Object.keys(resolverMappings)

typeKeys.forEach((typeKey) => {
  const type = resolverMappings[typeKey]
  const resolversKeysForType = Object.keys(type)
  result[typeKey] = {}

  resolversKeysForType.forEach((resolverKey) => {
    const resolverFn = type[resolverKey]
    output[typeKey][resolverKey] = wrapSingleResolverForMetrics(resolverFn)
  })
})

```

To be honest... I don't really know if that even improves it. Feel free to ignore it :)

Reply:
Ok, I am +1 on using `Object.keys()` so that `hasOwnProperty` check is not necessary. That would eliminate the `continue` statements.
But I like the `for in` loop. IMO more straightforward than the `.forEach` thing.